### PR TITLE
add report of loans made by the student

### PR DESCRIPTION
**Pull request by Magi1610**
This Pull Request integrates the Applicant Loan Report functionality into the system while introducing enhancements to the inventory and loan modules. Below are the detailed changes included:

Changes Summary
Applicant Loan Report
Added a new view: resources/views/loans/reportStudents.blade.php to display loans associated with applicants.
Updated LoanController to handle the generation of applicant-specific loan reports.
Updated routes (routes/web.php) to include new endpoints for loan reports.
Inventory Enhancements
Modified InventoryController to support filtering by an optional ignore-status parameter.
Adjusted the inventory report generation logic to include more robust filtering.
Added support for displaying inventory materials with improved grouping in resources/views/inventories/report.blade.php.
Table Adjustments
Updated table layout in loans/index.blade.php and reports/inventoryReport.blade.php for better data presentation:
Grouped materials by inventory with row spans.
Ensured materials and quantities are displayed correctly in reports.

### CODE REVIEW

File: app/Http/Controllers/InventoryController.php [hunk 1]
```diff
@@ -105,14 +105,19 @@ public function inventoryReport(Request $request)
         $status = $request->input('inventoryStatus');
         $startDate = $request->input('startDate');
         $endDate = $request->input('endDate');
+        $ignoreStatus = $request->input('ignoreStatus');
+
```
Comment (And-Y21) [inline app/Http/Controllers/InventoryController.php]: quitar el espacio

File: resources/views/inventories/report.blade.php [hunk 2]
```diff
@@ -17,11 +17,20 @@
                             <option value="no disponible">No disponible</option>
                         </select>
 
+                        <div class="form-check">
+                            <input type="checkbox" class="form-check-input" id="ignoreStatus" name="ignoreStatus">
+                            <label class="form-check-label" for="ignoreStatus">Ignorar estado (solo por fechas)</label>
+                        </div>
+
```
Comment (And-Y21) [inline resources/views/inventories/report.blade.php]: quitar los espacios del html

File: resources/views/inventories/report.blade.php [hunk 3]
```diff
@@ -34,15 +43,48 @@
...
+        endDateInput.addEventListener('change', validateDates);
+
+        form.addEventListener('submit', (event) => {
+            event.preventDefault();
+
+            if (!validateDates()) return;
+
+            const inventoryStatus = document.getElementById('inventoryStatus').value;
+            const ignoreStatus = document.getElementById('ignoreStatus').checked ? 1 :
+            0; // Enviar el valor del checkbox
```
Comment (And-Y21) [inline resources/views/inventories/report.blade.php]: quitar comentario

File: resources/views/loans/reportStudents.blade.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                            @foreach ($students as $student)
+                                <option value="{{ $student->id }}">{{ $student->name }} {{ $student->last_name }}
+                                </option>
+                            @endforeach
+                        </select>
+
+                        <label for="startDate" class="form-label mt-3">Fecha inicio(*)</label>
+                        <input type="date" id="startDate" name="startDate" class="form-control" required
+                            placeholder="Ingrese fecha inicio" />
+
```
Comment (And-Y21) [inline resources/views/loans/reportStudents.blade.php]: quitar los espacios

File: resources/views/loans/reportStudents.blade.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        form.addEventListener('submit', (event) => {
+            event.preventDefault();
+
+            if (!validateDates()) return;
+
+            const studentId = document.getElementById('studentId').value;
+            if (!studentId) {
+                alert("Por favor, selecciona un solicitante.");
+                return;
+            }
```
Comment (And-Y21) [inline resources/views/loans/reportStudents.blade.php]: Si ya tienes un required en el select no es necesaria la alerta

File: resources/views/loans/reportStudents.blade.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            const endDate = new Date(endDateInput.value);
+
+            if (startDate && endDate && startDate >= endDate) {
+                dateError.style.display = 'block';
+                return false;
+            } else {
+                dateError.style.display = 'none';
+                return true;
+            }
+        };
```
Comment (And-Y21) [inline resources/views/loans/reportStudents.blade.php]: Mejorar el codigo para que no sea if-else

File: resources/views/inventories/report.blade.php (hunk 3 above)
Comment (Magi1610) [reply resources/views/inventories/report.blade.php]: listo

File: resources/views/loans/reportStudents.blade.php (hunk 6 above)
Comment (Magi1610) [reply resources/views/loans/reportStudents.blade.php]: listo ya realice los cambios

**Pull request merged**